### ISSUE TITLE
@VirtualProperty "exp" does not play nice with @ExclusionPolicy("ALL")

### ISSUE THREAD

**Issue by JustBlackBird** [schmittjoh/serializer#746]
I'm trying to solve problem similar to #448.

I have `Department` class:

```php
class Department
{
    private $id;

    public function __construct($id)
    {
        $this->$id = $id;
    }

    public function getId()
    {
        return $this->id;
    }
}
```

If I define `User` class like this

```php
/**
 * @Serializer\ExclusionPolicy("ALL")
 * @Serializer\VirtualProperty(
 *     "dep",
 *     exp="object.getDepartment().getId()",
 *     options={@Serializer\SerializedName("dep")}
 * )
 */
class User
{
    private $id;
    private $department;

    public function __construct($id, Department $department)
    {
        $this->id = $id;
        $this->department = $department;
    }

    /**
     * @Serializer\VirtualProperty()
     * @Serializer\SerializedName("id")
     */
    public function getId()
    {
        return $this->id;
    }

    public function getDepartment()
    {
        return $this->department;
    }
}
```
then after serialization i get something like:
```json
{"id": 42}
```
_Notice there is no `dep` field._

At the same time, if I define `User` as this:
```php
/**
 * @Serializer\VirtualProperty(
 *     "dep",
 *     exp="object.getDepartment().getId()",
 *     options={@Serializer\SerializedName("dep")}
 * )
 */
class User
{
    /**
     * @Seializer\Exclude()
     */
    private $id;
    /**
     * @Seializer\Exclude()
     */
    private $department;

    public function __construct($id, Department $department)
    {
        $this->id = $id;
        $this->department = $department;
    }

    /**
     * @Serializer\VirtualProperty()
     * @Serializer\SerializedName("id")
     */
    public function getId()
    {
        return $this->id;
    }

    public function getDepartment()
    {
        return $this->department;
    }
}
```
I get
```json
{"id": 42, "dep": 12}
```
___

It seems that `@ExclusionPolicy("ALL")` somehow breaks virtual properties based on expression language.

Am I missing something?

**Comment by goetas** [comment]
thanks for the bug report, has been fixed in 1.6.2

**Comment by JustBlackBird** [comment]
Thank you for the fix 😄
